docs_src/getting-started/tools/Ch-CommandLineInterface.md

```diff
@@ -0,0 +1,104 @@
+
+The Edgex-CLI is also publish in snap store (Snapcraft.io). To install the EdgeX CLI snap run:
+
+```
+sudo snap install edgex-cli
+```
+
+You can also get the source-code and then build and install edgex-cli from the source.  Again, find the source code on [GitHub](https://github.com/edgexfoundry/edgex-cli).
+
+**For more information read see the tool's [README](https://github.com/edgexfoundry/edgex-cli/blob/master/README.md).**
+
+## What can the CLI do?
+
+The Edgex CLI covers most of the basic EdgeX Foundry APIs. Using the CLI, users can query all the available devices, profiles, services, events, readings etc. or query for a subset of them based on available flags – for example get devices with a specific name or get all events that belong to specific device.
+
+It also provides the means to create EdgeX objects such as devices, profiles, etc.  Creating can be accomplished in three different ways:  
+- By providing a file that contains the normal create object API request body
+- By using inline parameters
+- Or using interactive mode.
+
+Some of the request bodies could contain a lot parameters and writing all them in one single line may not be convenient.  For these cases, the interactive mode used/or not altogether with inline parameters could be more convenient.
+
+
+Updating EdgeX objects via the CLI can done in two ways:
+ - By providing a file that contains the update API request body that would normally be used in the API call
+ - Or by providing just the name of the object and, if the object with such name exists, enter into editor mode with the preloaded request body where you can change/edit just the fields that need to be modified. 
+
+The available CLI capabilities, commands and option flags are too many to share here.  Use –help with the CLI to get more information.
+
+## Example Usage
+
+By default, the CLI results are printed out in tabular format, but by providing –v flag the result is printed in plain json format.  Here are some common CLI command examples, using the EdgeX device as the object, to show how to query, create, update and delete EdgeX objects.  The same CLI command principals generally apply for the other EdgeX objects (such as profile, services, events, readings, etc.) 
+
+### List
+
+You can easily get all the available EdgeX devices or get a device with a specific name.
+
+``` bash
+edgex-cli device list 
+```
+
+![image](device-list.png)
+
+``` bash
+edgex-cli device list – name Random-Integer-Device
+```
+
+![image](device-list-by-name.png)
+
+``` bash
+edgex-cli device list – name Random-Integer-Device -v
+```
+
+![image](device-list-json.png)
+
+### Create
+Here is an example of how the CLI's interactive mode can be used together with inline parameters to create the main structure of a new create device request. All provided inline parameters are already populated and the user is left to populate the rest of the needed values. 
```

Comment:
s/to create the/to populate the
s/All provided inline parameters are already populated  /Note how all the inline provided parameters are pre-populated